### PR TITLE
example: secret as code

### PR DESCRIPTION
**Pull request by howieyuen**
related: https://github.com/KusionStack/kusion/issues/149#issuecomment-1311200340

### CODE REVIEW

File: base/examples/secret-as-code/project.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# The project basic info
+name: secret-as-code
+secret_stores:
```
Comment (SparkYuan) [inline base/examples/secret-as-code/project.yaml:3]: please add some comments here or add a user guide in our website to help users to understand how to config their secret stores

File: base/examples/secret-as-code/dev/main.k [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import base.pkg.kusion_models.kube.frontend
+
+# The application configuration in stack will overwrite 
+# the configuration with the same attribute in base.
+appConfiguration: frontend.Server {
+    annotations: {
+        "secret-store": "vault"
+        "foo": "ref+vault://secret/foo#/foo"
```
Comment (SparkYuan) [inline base/examples/secret-as-code/dev/main.k:8]: the format of this annotation is still confusing for a beginner, please add some comments to explain this format

File: base/examples/secret-as-code/project.yaml [hunk 3]
```diff
@@ -0,0 +1,11 @@
...
+  vault:
+    # address: http://127.0.0.1:8200
+    # auth_method: token
+    # token_file: /Users/yuanhao/.vault-token
```
Comment (SparkYuan) [inline base/examples/secret-as-code/project.yaml:7]: Are these comments meaningful or just for test？

File: base/examples/secret-as-code/project.yaml (hunk 1 above)
Comment (howieyuen) [reply base/examples/secret-as-code/project.yaml:3]: new user guide will be posted, after the feat PR and example case get merged.

File: base/examples/secret-as-code/project.yaml (hunk 3 above)
Comment (howieyuen) [reply base/examples/secret-as-code/project.yaml]: original idea is to example two authentication method, now I delete it, leave this in user guide

File: base/examples/secret-as-code/dev/main.k (hunk 2 above)
Comment (howieyuen) [reply base/examples/secret-as-code/dev/main.k:10]: nice catch, I will add it

**Review by markliby — APPROVED** [review]
lgtm

**Pull request merged**